src/lib.rs

```diff
@@ -1,90 +1,122 @@
-#![recursion_limit="128"]
-
 extern crate proc_macro;
-extern crate syn;
-#[macro_use]
-extern crate quote;
 
-use proc_macro::TokenStream;
-use syn::{Ident, DeriveInput};
+use quote::quote;
+
+use proc_macro2::{Span, TokenStream};
+use syn::{parse_macro_input, DeriveInput, Ident};
 
 struct Idents {
     item: Ident,
     collection: Ident,
-    derives: Option<Vec<Ident>>
+    derives: Option<Vec<Ident>>,
 }
 
 impl Idents {
-
-    fn new(ast: &DeriveInput) -> Result<Idents, String> {
-        let collection_name = attr_string_val(ast, "CollectionName").expect("Need [CollectionName=\"...\"]");
-        let derives = Idents::parse_derives(ast);
-
-        Ok(Idents{
-            item: ast.ident.clone(),
-            collection: Ident::from(collection_name),
-            derives: derives
+    fn new(input: &DeriveInput) -> Result<Idents, String> {
+        let collection_name =
+            attr_string_val(input, "CollectionName").expect("Need [CollectionName=\"...\"]");
+        let derives = Idents::parse_derives(input);
+
+        Ok(Idents {
+            item: input.ident.clone(),
+            collection: Ident::new(&collection_name, Span::call_site()),
+            derives: derives,
         })
     }
 
-    fn parse_derives(ast: &DeriveInput) -> Option<Vec<Ident>> {
-        match attr_string_val(ast, "CollectionDerives") {
+    fn parse_derives(input: &DeriveInput) -> Option<Vec<Ident>> {
+        match attr_string_val(input, "CollectionDerives") {
             Some(derives_str) => {
                 if derives_str.is_empty() {
-                    return None
+                    return None;
                 }
 
-                Some(derives_str.split(",").map(Ident::from).collect())
-            },
-            None => None
+                Some(
+                    derives_str
+                        .split(",")
+                        .map(|s| Ident::new(s.trim(), Span::call_site()))
```

Comment:
I had to add the trim() here, otherwise `#[CollectionDerives="Clone, Debug"]` would complain about deriving `" Debug"`. I think the modern way to do this would be to use a [Meta::List](https://docs.rs/syn/1.0.54/syn/enum.Meta.html) but I wasn't sure if this could be done without changing the API of this crate.